### ISSUE TITLE
ArangoDB exits without logs when executing basic group by query (Docker: 139 / SIGSEGV)

### ISSUE THREAD

**Issue by millii19** [arangodb/arangodb#12267]
## My Environment

* __ArangoDB Version__:        3.6.0, 3.6.4, 3.6.5
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Single Server
* __Deployment Strategy__:   Manual Start in Docker
* __Configuration__:               N/A
* __Infrastructure__:               own Laptop/PC
* __Operating System__:        Manjaro Lysia (20.0.3), Windows 10
* __Total RAM in your machine__:        16GB
* __Disks in use__:        SSD, HDD
* __Used Package__:              Docker - official Docker library and Windows Service

## Component, Query & Data

__Affected feature__:
AQL query using web interface and with arangojs driver

__AQL query (if applicable)__:
```
LET res = (FOR i in Customers
    FOR o IN 1..1 ANY i Customers2Order
    //FILTER o.Auftraggeber == i.Kunden
        COLLECT ort=i.Ort, kat=i.Klassifikation, mat=o.Material
        RETURN {
            ort,
            kat,
            mat
 })
RETURN res
```

__AQL explain (if applicable)__:
```
Query String (262 chars, cacheable: true):
 LET res = (FOR i in Customers
     FOR o IN 1..1 ANY i Customers2Order
     //FILTER o.Auftraggeber == i.Kunden
         COLLECT ort=i.Ort, kat=i.Klassifikation, mat=o.Material
         RETURN {
             ort,
             kat,
             mat
         })
 RETURN res

Execution plan:
 Id   NodeType                  Est.   Comment
  1   SingletonNode                1   * ROOT
 14   SubqueryStartNode            1     - LET res = ( /* subquery begin */
  3   EnumerateCollectionNode    440       - FOR i IN Customers   /* full collection scan */
  5   CalculationNode            440         - LET #10 = i.`Ort`   /* attribute expression */   /* collections used: i : Customers */
  6   CalculationNode            440         - LET #12 = i.`Klassifikation`   /* attribute expression */   /* collections used: i : Customers */
  4   TraversalNode               33         - FOR o  /* vertex */ IN 1..1  /* min..maxPathDepth */ ANY i /* startnode */  Customers2Order
  7   CalculationNode             33           - LET #14 = o.`Material`   /* attribute expression */
  8   CollectNode                 26           - COLLECT ort = #10, kat = #12, mat = #14   /* hash */
 13   SortNode                    26           - SORT ort ASC, kat ASC, mat ASC   /* sorting strategy: standard */
  9   CalculationNode             26           - LET #16 = { "ort" : ort, "kat" : kat, "mat" : mat }   /* simple expression */
 15   SubqueryEndNode             26     - RETURN  #16 ) /* subquery end */
 12   ReturnNode                  26     - RETURN res

Indexes used:
 By   Name   Type   Collection        Unique   Sparse   Selectivity   Fields        Ranges
  4   edge   edge   Customers2Order   false    false        99.99 %   [ `_to` ]     base INBOUND
  4   edge   edge   Customers2Order   false    false         3.09 %   [ `_from` ]   base OUTBOUND

Traversals on graphs:
 Id  Depth  Vertex collections  Edge collections                  Options                                  Filter / Prune Conditions
 4   1..1                       Customers2Order, Customers2Order  uniqueVertices: none, uniqueEdges: path                           

Optimization rules applied:
 Id   RuleName
  1   move-calculations-up
  2   move-calculations-up-2
  3   move-calculations-down
  4   splice-subqueries
```

__Dataset__:
[Dump](https://gist.github.com/millii19/38529049901600a8fd32439b43670fc3)

__Size of your Dataset on disk__:
~9MB

__Replication Factor & Number of Shards (Cluster only)__:
N/A

## Steps to reproduce

1. Import database dump
2. execute above query
3. database will crash shortly after execution

__Problem__:
Docker container exits with error code 139. On windows the arangodb service just stops and on a native install on linux the database also stops without any logs.
This often happens a bit after the database executes the query and the result is visible in the web interface.

Weirdly, when using the filter that is commented in the above query, it works. Please note that the filter is unnecessary, because the edge collection was built using the same filter.

__Expected result__:
returns something and database doesn't quit

**Comment by millii19** [comment]
Oh, I forgot to mention that it works with ArangoDB 3.5.5

**Comment by dothebart** [comment]
hm, there should be a vertex and an edge collection, you just posted one collection?

**Comment by millii19** [comment]
The dump should have 8 files, it only displays one on the website though, scroll to the bottom. Or download the zip

**Comment by dothebart** [comment]
The fix will be available with ArangoDB 3.6.6. 
Please follow the official release channels for notification.
